app/lib/analyzer/backend.dart

```diff
@@ -63,10 +63,7 @@ class AnalysisBackend {
   /// Stores the analysis, and either creates or updates its parent
   /// [PackageAnalysis] and [PackageVersionAnalysis] records.
   Future storeAnalysis(Analysis analysis) async {
-    assert(analysis.id == null);
-    await db.commit(inserts: [analysis]);
-    assert(analysis.id != null);
-
+    analysis.id ??= new DateTime.now().millisecondsSinceEpoch;
```

Comment:
Something like this [the current new code]? I'm not sure if it is supposed to be called in our outside of the transaction.

Btw. with the `DateTime`-based solution, I was referring to this API:
https://api.dartlang.org/stable/1.24.2/dart-core/DateTime/millisecondsSinceEpoch.html

with that, `new DateTime.now().millisecondsSinceEpoch == new DateTime.now().toUtc().millisecondsSinceEpoch`